src/Native/System.Security.Cryptography.Native/pal_rsa.cpp

```diff
@@ -12,3 +31,122 @@ extern "C" RSA* DecodeRsaPublicKey(const unsigned char* buf, int32_t len)
 
     return d2i_RSAPublicKey(nullptr, &buf, len);
 }
+
+extern "C" int32_t RsaPublicEncrypt(int32_t flen, const uint8_t* from, uint8_t* to, RSA* rsa, int32_t useOaepPadding)
+{
+    int padding = useOaepPadding ? RSA_PKCS1_OAEP_PADDING : RSA_PKCS1_PADDING;
+    return RSA_public_encrypt(flen, from, to, rsa, padding);
+}
+
+extern "C" int32_t RsaPrivateDecrypt(int32_t flen, const uint8_t* from, uint8_t* to, RSA* rsa, int32_t useOaepPadding)
+{
+    int padding = useOaepPadding ? RSA_PKCS1_OAEP_PADDING : RSA_PKCS1_PADDING;
+    return RSA_private_decrypt(flen, from, to, rsa, padding);
+}
+
+extern "C" int32_t RsaSize(RSA* rsa)
+{
+    return RSA_size(rsa);
+}
+
+extern "C" int32_t RsaGenerateKeyEx(RSA* rsa, int32_t bits, BIGNUM* e)
+{
+    return RSA_generate_key_ex(rsa, bits, e, nullptr);
+}
+
+extern "C" int32_t RsaSign(int32_t type, const uint8_t* m, int32_t mlen, uint8_t* sigret, int32_t* siglen, RSA* rsa)
+{
+    if (!siglen)
+    {
+        assert(false);
+        return 0;
+    }
+
+    unsigned int unsignedSigLen = 0;
+    int32_t ret = RSA_sign(type, m, UnsignedCast(mlen), sigret, &unsignedSigLen, rsa);
+    assert(unsignedSigLen <= INT32_MAX);
+    *siglen = static_cast<int32_t>(unsignedSigLen);
+    return ret;
+}
+
+extern "C" int32_t RsaVerify(int32_t type, const uint8_t* m, int32_t mlen, uint8_t* sigbuf, int32_t siglen, RSA* rsa)
+{
+    return RSA_verify(type, m, UnsignedCast(mlen), sigbuf, UnsignedCast(siglen), rsa);
+}
+
+extern "C" void GetRsaParameters(const RSA* rsa,
+                                 BIGNUM** n,
+                                 BIGNUM** e,
+                                 BIGNUM** d,
+                                 BIGNUM** p,
+                                 BIGNUM** dmp1,
+                                 BIGNUM** q,
+                                 BIGNUM** dmq1,
+                                 BIGNUM** iqmp)
+{
+    if (!rsa || !n || !e || !d || !p || !dmp1 || !q || !dmq1 || !iqmp)
+    {
+        assert(false);
+        return;
```

Comment:
If we return here, there was an invalid parameter passed in. If this was invoked by managed code, none of the 'out' parameters can be NULL, can they? And the managed code already has a check for an invalid 'rsa' pointer before calling this.
So really this guard is for someone else invoking this code, and because it's a good practice to always check pointers before de-referencing them.

Is what you are looking for, something like this?

```C++
if (!rsa || !n || !e || !d || !p || !dmp1 || !q || !dmq1 || !iqmp)
{
    assert(false);
    if (n)
        *n = nullptr;
    if (e)
        *e = nullptr;
    if (d)
        *d = nullptr;
    if (p)
        *p = nullptr;
    if (dmp1)
        *dmp1 = nullptr;
    if (q)
        *q = nullptr;
    if (dmq1)
        *dmq1 = nullptr;
    if (iqmp)
        *iqmp = nullptr;
    return;
}
*n = rsa->n;
*e = rsa->e;
*d = rsa->d;
*p = rsa->p;
*dmp1 = rsa->dmp1;
*q = rsa->q;
*dmq1 = rsa->dmq1;
*iqmp = rsa->iqmp;
```